From 565756e13130d32eef93f93f93be2fff379cb0f8 Mon Sep 17 00:00:00 2001
From: Sebastian Wick <sebastian.wick@redhat.com>
Date: Mon, 11 Aug 2025 11:39:02 +0200
Subject: [PATCH] wayland/transaction: Avoid crash on surfaces without roles
 with

...a time constraint.

When the surface has no role assigned yet, it doesn't have an actor.
Ideally we would still manage to schedule the commits, but for now we
just ignore the time constraint.

Related: https://gitlab.gnome.org/GNOME/mutter/-/issues/4108
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4566>
---
 src/wayland/meta-wayland-transaction.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/wayland/meta-wayland-transaction.c b/src/wayland/meta-wayland-transaction.c
index d0ca29ec25f..0419663a53f 100644
--- a/src/wayland/meta-wayland-transaction.c
+++ b/src/wayland/meta-wayland-transaction.c
@@ -503,9 +503,17 @@ meta_wayland_transaction_commit (MetaWaylandTransaction *transaction)
    */
   if (max_time_us)
     {
-      MetaSurfaceActor *actor = meta_wayland_surface_get_actor (max_time_surface);
-      ClutterFrameClock *frame_clock =
-        clutter_actor_pick_frame_clock (CLUTTER_ACTOR (actor), NULL);
+      MetaSurfaceActor *actor;
+      ClutterFrameClock *frame_clock = NULL;
+
+      /* When the surface has no role assigned yet, it doesn't have an actor.
+       * Ideally we would still manage to schedule the commits, but for now we
+       * just ignore the time constraint.
+       * See: https://gitlab.gnome.org/GNOME/mutter/-/issues/4108
+       */
+      actor = meta_wayland_surface_get_actor (max_time_surface);
+      if (actor)
+        frame_clock = clutter_actor_pick_frame_clock (CLUTTER_ACTOR (actor), NULL);
 
       if (frame_clock)
         {
-- 
GitLab

